### ISSUE TITLE
Use of annotation 'scheduler.alpha.kubernetes.io/critical-pod' in DaemonSet resource descriptor without effect

### ISSUE THREAD

**Issue by dhaubenr** [ibm-s390-cloud/k8s-cex-dev-plugin#3]
When deploying the 'cex-plugin-daemonset' DaemonSet on an OCP 4.9.x cluster (which itself is based on Kubernetes 1.22+) using the sample provided in this repo, the following warning is displayed:

```
W1222 13:03:06.893635   65445 warnings.go:70] spec.template.metadata.annotations[scheduler.alpha.kubernetes.io/critical-pod]: non-functional in v1.16+; use the "priorityClassName" field instead
daemonset.apps/cex-plugin-daemonset created
```

The DaemonSet descriptor should be adapted to make use of the 'prorityClassName' pod template attribute instead, using one of the two priority classes defined in Kubernetes out-of-the-box: 'system-cluster-critical' or 'system-node-critical'.

For more information see also:
- https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/
- https://kubernetes.io/docs/tasks/administer-cluster/guaranteed-scheduling-critical-addon-pods/

**Comment by hfreude** [comment]
Yes, this is true. Will update the sample deployment and docu.

**Comment by jschintag** [comment]
This issue has been fixed in commit [4702109e76891e4d3dd9f9a629f4017274fc346d](https://github.com/ibm-s390-cloud/k8s-cex-dev-plugin/commit/4702109e76891e4d3dd9f9a629f4017274fc346d)
